app/services/get_products_service.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class GetProductsService < BaseService
+  def initialize(category_name)
+    super()
+    @category_name = category_name.downcase
+    @white_list = [
+      I18n.t('category.all', locale: 'en'),
+      I18n.t('category.all', locale: 'vn')
+    ]
+  end
+
+  def call
+    validate
+    return self if error?
+
+    @result = fetch_products
+  end
+
+  private
+
+  attr_reader :category_name, :white_list
+
+  def fetch_products
+    query = Category.joins(:products)
+                    .left_outer_joins(products: :tag)
+                    .select('categories.name AS name, products.name AS product_name,
+                                  products.price AS product_price, products.thumbnail AS product_thumbnail,
+                                  products.id AS product_id, tags.name AS tag_name, tags.color AS tag_color')
+
+    if white_list.include?(category_name)
+      query
+    else
+      query.where(categories: { name: category_name })
+    end
+  end
+
+  def validate
+    return unless category_name.nil? || Category.where(name: category_name).blank?
```

Comment:
this have to check both condition ? why don't check each of them and then fail fast for each case
and I think this is similar to what we already done, should review ur code